tools/_functions.sh

```diff
@@ -473,3 +473,24 @@ function join_by {
         printf %s "$f" "${@/#/$d}"
     fi
 }
+
+# This function checks if the flag "--as-precommit" was set, and if so,
+# runs the specified pre-commit hook against all changed files
+function run_as_precommit {
+    local command="$1"
+    shift
+
+    for arg in "$@"; do
+        if [[ "$arg" == "--as-precommit" ]]; then
+            FILES=$(git diff --cached --name-only --diff-filter=ACMR | sed 's| |\\ |g')
+            [ -z "$FILES" ] && exit 0
+
+            while IFS= read -r file; do
+                eval "$command \"$file\""
+                git add "$file"
```

Comment:
I am a bit worried that the changes may not be noticed by the committer. For testing, I have modified a ts file and added some code with bad indentation, and after writing `git commit`, the commit message editor opened and the indentation in the file was fixed, without informing me about it:
![image](https://github.com/user-attachments/assets/b38e9109-4e40-4130-87d8-5539b53f3f25)

Personally, I would prefer to not automatically add the files back, so that I can know what was changed by pre-commit. Although I would be fine with it if there was at least some kind of feedback that a file has been changed.